### PR TITLE
Student: Comment on responses: Fix put document error #9034

### LINKED ISSUE #9034: Student: Comment on responses: Fix put document error
Put comment search document fails silently after adding new feedback participant comment because of null comment id.

### PR DESCRIPTION
**Pull request by nidhi98gupta**
Fixes #9034

### CODE REVIEW

**Comment by nidhi98gupta** [comment]
@xpdavid Put comment document was failing silently because comment id was null. Please review.

File: src/main/java/teammates/ui/controller/FeedbackSubmissionEditSaveAction.java [hunk 1]
```diff
@@ -307,7 +307,7 @@ private void saveNewCommentsByFeedbackParticipant(List<FeedbackResponseCommentAt
             throws EntityDoesNotExistException {
         for (FeedbackResponseCommentAttributes frc : commentsToSave) {
             try {
-                logic.createFeedbackResponseComment(frc);
```
Comment (xpdavid) [inline src/main/java/teammates/ui/controller/FeedbackSubmissionEditSaveAction.java]: Should add a test for this.

**Comment by nidhi98gupta** [comment]
> Will the test fail before the fix?

Yes

**Comment by xpdavid** [comment]
Should not have exception now.

image

File: src/test/java/teammates/test/cases/BaseComponentTestCase.java [hunk 2]
```diff
@@ -123,4 +125,26 @@ protected String doPutDocuments(DataBundle dataBundle) {
...
 
+    /*
+     * Verifies that search results match with expected output.
+     * Compares the text for each comment as it is unique.
+     *
+     * @param actual the results from the search query.
+     * @param expected the expected results for the search query.
+     */
+    protected static void verifySearchResults(FeedbackResponseCommentSearchResultBundle actual,
+                                            FeedbackResponseCommentAttributes... expected) {
```
Comment (damithc) [inline src/test/java/teammates/test/cases/BaseComponentTestCase.java]: should be 8-space indentation?

**Pull request merged**